saleor/tests/e2e/orders/discounts/test_order_products_on_catalog_promotion_and_order_promotion.py

```diff
@@ -0,0 +1,463 @@
+    expected_product1_unit_gross_price = round(
+        product1_variant_price
+        - (product1_variant_price * (catalog_promotion_value / 100)),
+        2,
+    )
+    expected_product1_unit_tax_price = round(
+        (expected_product1_unit_gross_price * country_tax_rate)
+        / (100 + country_tax_rate),
+        2,
+    )
+    assert order_line1["variant"]["id"] == product_variant_id_1
+    assert order_line1["unitDiscountReason"] == f"Promotion: {catalog_promotion_id}"
+    assert order_line1["unitDiscountValue"] == catalog_promotion_value
+    assert order_line1["unitDiscountType"] == "PERCENTAGE"
+    assert order_line1["unitDiscount"]["amount"] == round(
+        product1_variant_price * (catalog_promotion_value / 100), 2
+    )
+    assert (
+        order_line1["unitPrice"]["gross"]["amount"]
+        == expected_product1_unit_gross_price
+    )
+    assert order_line1["unitPrice"]["tax"]["amount"] == expected_product1_unit_tax_price
+    assert (
+        order_line1["unitPrice"]["gross"]["amount"]
+        != order_line1["undiscountedUnitPrice"]["gross"]["amount"]
+    )
+
+    expected_product2_unit_gross_price = round(
+        product2_variant_price
+        - (product2_variant_price * (catalog_promotion_value / 100)),
+        2,
+    )
+    expected_product2_unit_tax_price = round(
+        (expected_product2_unit_gross_price * country_tax_rate)
+        / (100 + country_tax_rate),
+        2,
+    )
+    assert order_line2["variant"]["id"] == product_variant_id_2
+    assert order_line2["unitDiscountReason"] == f"Promotion: {catalog_promotion_id}"
+    assert order_line2["unitDiscountValue"] == catalog_promotion_value
+    assert order_line2["unitDiscountType"] == "PERCENTAGE"
+    assert order_line2["unitDiscount"]["amount"] == round(
+        product2_variant_price * (catalog_promotion_value / 100), 2
+    )
+    assert (
+        order_line2["unitPrice"]["gross"]["amount"]
+        == expected_product2_unit_gross_price
+    )
+    assert order_line2["unitPrice"]["tax"]["amount"] == expected_product2_unit_tax_price
+    assert (
+        order_line2["unitPrice"]["gross"]["amount"]
+        != order_line2["undiscountedUnitPrice"]["gross"]["amount"]
+    )
+
+    # Assert subtotal:
+    subtotal_gross = round(
+        (expected_product1_unit_gross_price + expected_product2_unit_gross_price), 2
+    )
+    assert order["order"]["subtotal"]["gross"]["amount"] == subtotal_gross
+    subtotal_tax = round(
+        (subtotal_gross * country_tax_rate) / (100 + country_tax_rate),
+        2,
+    )
+    assert order["order"]["subtotal"]["tax"]["amount"] == subtotal_tax
+
+    # Step 3 - Update quantity for fist product
+    input = {"quantity": 5}
+    order_line_update(e2e_staff_api_client, order_line1["id"], input)
+
+    # Step 4 - Update quantity for second product
+    input = {"quantity": 3}
+    order = order_line_update(e2e_staff_api_client, order_line2["id"], input)
+
+    # Assert undiscounted total is over 500$
```

Comment:
It's not clear why above 500. Maybe it should be calculated?

Reply:
Here I want to check if the total met the order promotion conditions I calculate the expected undiscounted total later with shipping and compare it with the API results. I added a variable `order_promotion_total_predicate` and used it to create an order promotion and here.